bigquery/cloud-client/src/main/java/com/example/bigquery/UpdateDatasetAccess.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+// [START bigquery_update_dataset_access]
+import com.google.cloud.bigquery.Acl;
+import com.google.cloud.bigquery.Acl.Role;
+import com.google.cloud.bigquery.Acl.User;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetInfo;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UpdateDatasetAccess {
+
+  public static void runUpdateDatasetAccess() {
+    // TODO(developer): Replace these variables before running the sample.
+    String datasetName = "my-dataset-name";
+    updateDatasetAccess(datasetName);
+  }
+
+  public static void updateDatasetAccess(String datasetName) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+    Dataset dataset = bigquery.getDataset(datasetName);
+    List<Acl> beforeAcls = dataset.getAcl();
+
+    // Make a copy of the ACLs so that they can be modified.
```

Comment:
This is somewhat misleading - you aren't making a copy of the ACLs, you are making a copy of the list (which still has references to the original ACL objects). The references in the list still point to the same objects. 

Also, there's no reason to retain a link to the original list, because it doesn't get used anywhere. 

Something like this might be better instead:
```suggestion
  // For more information on the types of ACLs available see: 
  // https://link.to.acl.docs
  // Create a new ACL granting the READER role to "sample.bigquery.dev@gmail.com"
  Acl newEntry = Acl.of(new Acl.User("sample.bigquery.dev@gmail.com"), Role.READER);
   
   // Get a copy of the ACLs list from the dataset and append the new entry
   ArrayList<Acl> acls = new ArrayList<>(dataset.getAcl());
   acls.add(newEntry);
   
   // Then down below add a comment like "Update the dataset to use the new ACLs".
```